### PR TITLE
Fix several bugs

### PR DESCRIPTION
**Pull request by DenysZP**
Fix placeholder vector image
Fix popup on several devices
Fix search issue
Fix category pagination issue
Fix image recycling issue

### CODE REVIEW

File: app/src/main/java/com/shopapp/ui/category/contract/CategoryContract.kt [hunk 1]
```diff
@@ -16,10 +16,10 @@ class CategoryPresenter @Inject constructor(private val categoryUseCase: Categor
 
         categoryUseCase.execute(
             {
-                if (it.productList.isNotEmpty()) {
-                    view?.showContent(it.productList)
-                } else if (paginationValue == null) {
+                if (paginationValue == null && it.productList.isEmpty()) {
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/category/contract/CategoryContract.kt]: Add test for paginationValue == null and it.productList not empty

File: app/src/main/java/com/shopapp/ui/search/contract/SearchContract.kt [hunk 2]
```diff
@@ -20,7 +20,13 @@ class SearchPresenter @Inject constructor(private val searchUseCase: SearchUseCa
             return
         }
         searchUseCase.execute(
-            { view?.showContent(it) },
+            {
+                if (paginationValue == null && it.isEmpty()) {
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/search/contract/SearchContract.kt]: Add tests for 4 cases

File: app/src/main/java/com/shopapp/ui/search/SearchWithCategoriesFragment.kt [hunk 3]
```diff
@@ -59,7 +59,7 @@ class SearchWithCategoriesFragment : Fragment(), SearchToolbar.SearchToolbarList
     }
 
     override fun onQueryChanged(query: String) {
-        if (searchFragment.isVisible) {
+        if (searchFragment.isVisible || query.isBlank()) {
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/search/SearchWithCategoriesFragment.kt]: Add test for query is not blank

File: app/src/main/java/com/shopapp/ui/order/list/contract/OrderListContract.kt [hunk 4]
```diff
@@ -15,7 +15,11 @@ class OrderListPresenter @Inject constructor(private val orderListUseCase: Order
 
         orderListUseCase.execute(
             {
-                view?.showContent(it)
+                if (paginationValue == null && it.isEmpty()) {
```
Comment (plzen) [inline app/src/main/java/com/shopapp/ui/order/list/contract/OrderListContract.kt]: Add tests for all 4 states
